### PR TITLE
[wasi] Update wasi-sdk to version 21.

### PR DESCRIPTION
**Pull request by vargaz**
Contributes to https://github.com/dotnet/runtime/issues/98315

### CODE REVIEW

**Comment by vargaz** [comment]
Re: https://github.com/dotnet/runtime/pull/98120

**Comment by pavelsavara** [comment]
cc @lewing

File: src/ubuntu/20.04/webassembly/Dockerfile [hunk 1]
```diff
@@ -49,7 +49,7 @@ RUN curl -sSL https://netcorenativeassets.blob.core.windows.net/resource-package
     && chmod +x /usr/local/bin/v8
 
 # Install Wasi toolchain
-ENV WASI_SDK_VERSION=20
+ENV WASI_SDK_VERSION=21
```
Comment (akoeplinger) [inline src/ubuntu/20.04/webassembly/Dockerfile:52]: I don't think we use this container anymore actually. We could probably delete it.

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
Note that we're currently pinning an older version of the container here: https://github.com/dotnet/runtime/blob/4f099091aa1dbcc3db0043241f3a4e81ef5abfdd/eng/pipelines/common/templates/pipeline-with-resources.yml#L88-L96

**Pull request merged**